boards/FSM/Src/Io/Io_FlowMeter.c

```diff
@@ -1,20 +1,91 @@
 #include <assert.h>
 #include "main.h"
-#include "Io_SharedPwmInput.h"
+#include "Io_SharedFrequencyOnlyPwmInput.h"
 #include "Io_FlowMeter.h"
 
-static struct PwmInput *  primary_flow_meter_pwm_input;
-static TIM_HandleTypeDef *primary_flow_meter_htim;
+#define PRIMARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_1
+#define SECONDARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_2
+
+extern TIM_HandleTypeDef htim4;
+
+static struct FrequencyOnlyPwmInput *primary_flow_meter_pwm_input,
```

Comment:
I think it'd make the code more readable if we group the variables in a struct 
```
struct FlowMeter_Context
{
  struct FreqOnlyPwmInput* pwm_input;
  size_t elapsed_tick;
  TIM_HandleTypeDef* timer_handler
}
```

Reply:
I followed what you suggested here excluding the timer_handler. I went with `static TIM_HandleTypeDef * flow_meters_htim` instead as both flow meters should share the same timer handle.

Reply:
Fixed.